lang/types/value.go

```diff
@@ -967,10 +1080,11 @@ func (obj *FuncValue) Value() interface{} {
 			// that something went horribly wrong, aka a panic...
 			panic(fmt.Sprintf("function panic: %+v", err))
 		}
-		return []reflect.Value{reflect.ValueOf(result.Value())} // only one result
+		out := reflect.New(v.Type().Out(0))
+		result.Into(out)
+		return []reflect.Value{out} // only one result
 	}
-	val := reflect.MakeFunc(typ, fn)
-	return val.Interface()
+	v.Set(reflect.MakeFunc(v.Type(), fn))
```

Comment:
I'm not actually 100% sure how the FuncValue type works so this is mostly a guess. It does pass tests though, assuming there are tests for this